Protractor tests standard names

satwik97 opened the pull request:
Fix for the issue #1579 
Naming Standard in fileName.js

    name: user1_fileName
    email: user1@fileName.com
    name: user2_fileName
    email: user2@fileName.com

jacobdavis11 commented on core/tests/protractor/privileges.js:
```diff
@@ -26,27 +26,29 @@ var workflow = require('../protractor_utils/workflow.js');
-    users.createUser('alice@example.com', 'Alice');
-    users.createUser('bob@example.com', 'Bob');
-    users.createUser('eve@example.com', 'Eve');
+    users.createUser('user1@privileges.com', 'user1_privileges');
```
Please keep the original alice, bob and eve here as their names fit the subsequent roles.

jacobdavis11 commented on core/tests/protractor/publicationAndGallery.js:
```diff
@@ -37,23 +37,27 @@ describe('Gallery view', function() {
-    users.createUser('celebrimbor@example.com', 'Celebrimbor');
-    users.createUser('earendil@example.com', 'Earendil');
-
-    users.login('feanor@exmple.com');
```
Please keep the existing names (with the new suffixes) in this file as they are thematic.

jacobdavis11 commented on core/tests/protractor/embedding.js:
```diff
@@ -62,8 +62,8 @@ describe('Embedding', function() {
 
-    users.createUser('embedder@example.com', 'Embedder');
-    users.login('embedder@example.com', true);
+    users.createUser('embedder@Embedding.com', 'Embedder');
```
The name also needs updating.

satwik97 replied:
Should I name it Embedder or
Embedder_embedding
or
user1_embedding
or
alice_embedding

jacobdavis11 replied:
user1 seems to make the most sense here.

satwik97 replied:
Done

jacobdavis11 commented on core/tests/protractor/explorationRating.js:
```diff
@@ -44,9 +44,9 @@ describe('Gallery view', function() {
   it('should display ratings on exploration when minimum ratings have been ' +
      'submitted', function() {
-    users.createUser('rating@example.com', 'Rating');
+    users.createUser('rating@explorationRating.com', 'Rating');
```
The name also needs updating.

jacobdavis11 commented on core/tests/protractor/fallbacks.js:
```diff
@@ -28,8 +28,8 @@ var player = require('../protractor_utils/player.js');
      'disallow editing of fallbacks in read-only explorations', function() {
-    users.createUser('fallback@example.com', 'fallback');
-    users.login('fallback@example.com');
+    users.createUser('user1@Fallbacks.com', 'user1');
```
The name also needs updating.

jacobdavis11 commented on core/tests/protractor/stateEditor.js:
```diff
@@ -28,8 +28,8 @@ var player = require('../protractor_utils/player.js');
   it('should display plain text content', function() {
-    users.createUser('user1@example.com', 'user1');
-    users.login('user1@example.com');
+    users.createUser('user1@stateEditor.com', 'user1');
```
The name also needs updating.

satwik97 replied:
The tests ran fine , however I forgot to add _filename to username

jacobdavis11 commented on core/tests/protractor/userManagement.js:
```diff
@@ -24,16 +24,19 @@ var workflow = require('../protractor_utils/workflow.js');
   it('should create users', function() {
-    users.createUser('ordinaryuser@example.com', 'ordinaryuser');
+    users.createUser(
+      'ordinaryuser@userManagement.com', 'ordinaryuser_userManagement');
```
I would use camelcase for the whole thing here.

satwik97 replied:
So should it be ordinaryUserUserManagement or ordinaryUserManagement ,
Do I have to  use camelcase in this particular case or in all of the usernames ?

jacobdavis11 replied:
I think ordinaryUserManagement is cleaner here. Please use camelcase in all usernames.

jacobdavis11 commented on core/tests/protractor/fallbacks.js:
```diff
@@ -59,7 +59,7 @@ describe('Fallbacks editor', function() {
       // allow the second user to modify the exploration.
       users.logout();
-      users.login('user7@example.com');
+      users.login('user7@Fallbacks.com');
```
Email should be lowercase.

jacobdavis11 commented:
Thanks! I made some comments.

seanlip commented:
@satwik97, @jacobdavis11 -- some quick notes:
- @satwik97, did you actually run the e2e tests? I thought all usernames needed to be made up of lowercase letters only, so I'm a bit surprised these tests are still passing.
- Shouldn't emails generally only contain lowercase alphanumeric characters (including digits)? I refer to things like `..@userManagement.com`.

satwik97 commented:
Hi @seanlip , my fault, I just realized that I just  run_presubmit_check.sh before pushing and missed run_e2e_tests.sh . Sorry for the wrong comment , tests are actually failing.
@jacobdavis11 underscores are not allowed in the username in oppia so we'll have to use camelCase only

seanlip commented:
Note that uppercase letters aren't used in usernames, either.

jacobdavis11 commented:
Using camelcase instead of underscores sounds good. I just tried upper case usernames and they seem fine, so I think we can keep those.

satwik97 commented:
@jacobdavis11 I've addressed all the comments and I've used camelCase for all usernaNames and emails with multiple words (like user1@editorAndPlayer.com) to make the email more readable.

jacobdavis11 commented:
Looks good. Thanks!

seanlip commented:
Closing this -- thanks, @satwik97!

@jacobdavis11 -- should the instructions in the wiki (re the e2e tests) be modified accordingly to explain the email/username convention?

jacobdavis11 commented:
Good idea, I've updated the wiki with the new convention.
